example/commitpr/main.go

```diff
@@ -135,7 +139,25 @@ func pushCommit(ref *github.Reference, tree *github.Tree) (err error) {
 	date := time.Now()
 	author := &github.CommitAuthor{Date: &github.Timestamp{Time: date}, Name: authorName, Email: authorEmail}
 	commit := &github.Commit{Author: author, Message: commitMessage, Tree: tree, Parents: []*github.Commit{parent.Commit}}
-	newCommit, _, err := client.Git.CreateCommit(ctx, *sourceOwner, *sourceRepo, commit)
+	opts := github.CreateCommitOptions{}
+	if *privateKey != "" {
+		armoredBlock, e := os.ReadFile(*privateKey)
+		if e != nil {
+			return e
+		}
+		keyring, e := openpgp.ReadArmoredKeyRing(bytes.NewReader(armoredBlock))
+		if e != nil {
+			return e
+		}
+		if len(keyring) != 1 {
+			return errors.New("expected exactly one key in the keyring")
+		}
+		key := keyring[0]
+		opts.Signer = github.MessageSignerFunc(func(w io.Writer, r io.Reader) error {
+			return openpgp.ArmoredDetachSign(w, key, r, nil)
+		})
```

Comment:
TIL that you can wrap a single `func` in a type cast that can then be used as a receiver in order to satisfy an interface...
so cool, @WillAbides - thank you!